src/transport/SessionHandle.h

```diff
@@ -55,6 +58,8 @@ class SessionHandle
     void SetFabricIndex(FabricIndex fabricId) { mFabric = fabricId; }
     void SetGroupId(GroupId groupId) { mGroupId.SetValue(groupId); }
 
+    SubjectDescriptor GetSubjectDescriptor() const;
```

Comment:
Should this return by value or by reference?

Reply:
I left it by value here, it's copied (as a snapshot) then used by reference after. But only for the duration of an interaction.

Reply:
It could probably be by reference also. I'm agnostic at the moment.